lib/ts/recipe/multifactorauth/index.ts

```diff
@@ -0,0 +1,148 @@
+        const user = await getUser(userId, ctx);
+        if (!user) {
+            throw new Error("UKNKNOWN_USER_ID");
+        }
+
+        return Recipe.getInstanceOrThrowError().recipeInterfaceImpl.getFactorsSetupForUser({
+            tenantId,
+            user,
+            userContext: ctx,
+        });
+    }
+
+    static async isAllowedToSetupFactor(
+        session: SessionContainerInterface,
+        factorId: string,
+        userContext?: Record<string, any>
+    ) {
+        let ctx = getUserContext(userContext);
+        const user = await getUser(session.getUserId(), ctx);
+        if (!user) {
+            throw new Error("UKNKNOWN_USER_ID");
+        }
+        const factorsSetup = await Recipe.getInstanceOrThrowError().recipeInterfaceImpl.getFactorsSetupForUser({
+            user,
+            tenantId: session.getTenantId(),
+            userContext: ctx,
+        });
+        const mfaClaimValue = await session.getClaimValue(MultiFactorAuthClaim, ctx);
+        const completedFactors = mfaClaimValue?.c ?? {};
+        const defaultMFARequirementsForUser = await Recipe.getInstanceOrThrowError().recipeInterfaceImpl.getDefaultRequiredFactorsForUser(
+            {
+                user,
+                tenantId: session.getTenantId(),
+                userContext: ctx,
+            }
+        );
+
+        const tenantInfo = await Multitenancy.getTenant(session.getTenantId(), userContext);
+        const defaultMFARequirementsForTenant: string[] = tenantInfo?.defaultRequiredFactorIds ?? [];
+        const requirements = await Recipe.getInstanceOrThrowError().recipeInterfaceImpl.getMFARequirementsForAuth({
+            user,
+            accessTokenPayload: session.getAccessTokenPayload(),
+            tenantId: session.getTenantId(),
+            factorsSetUpForUser: factorsSetup,
+            defaultRequiredFactorIdsForUser: defaultMFARequirementsForUser,
+            defaultRequiredFactorIdsForTenant: defaultMFARequirementsForTenant,
+            completedFactors,
+            userContext: ctx,
+        });
+        return Recipe.getInstanceOrThrowError().recipeInterfaceImpl.isAllowedToSetupFactor({
+            session,
+            factorId,
+            completedFactors,
+            mfaRequirementsForAuth: requirements,
+            factorsSetUpForUser: factorsSetup,
+            defaultRequiredFactorIdsForUser: defaultMFARequirementsForUser,
+            defaultRequiredFactorIdsForTenant: defaultMFARequirementsForTenant,
+            userContext: ctx,
+        });
+    }
+
+    static async markFactorAsCompleteInSession(
+        session: SessionContainerInterface,
+        factorId: string,
+        userContext?: Record<string, any>
+    ) {
+        return Recipe.getInstanceOrThrowError().recipeInterfaceImpl.markFactorAsCompleteInSession({
+            session,
+            factorId,
+            userContext: getUserContext(userContext),
+        });
+    }
+
+    static async addToDefaultRequiredFactorsForUser(
```

Comment:
Since the get function takes a tenantId, this should also take it.. If this does not need it, and we don't want tyo allow users to be able to customise to store the info on a tenant and user basis, then the get function should also not take a tenant id anymore?